python/plugins/grassprovider/Grass7Algorithm.py

```diff
@@ -139,11 +140,16 @@ def __init__(self, descriptionfile):
 
         # Use the ext mechanism
         name = self.name().replace('.', '_')
+        self.module = None
         try:
-            self.module = importlib.import_module(
-                'grassprovider.ext.{}'.format(name))
-        except ImportError:
-            self.module = None
+            extpath = Path(self.descriptionFile).parents[1].joinpath('ext', name + '.py')
+            # this check makes it a bit faster
+            if extpath.exists():
+                spec = importlib.util.spec_from_file_location('grassprovider.ext.' + name, extpath)
+                self.module = importlib.util.module_from_spec(spec)
+                spec.loader.exec_module(self.module)
+        except Exception:
```

Comment:
can we avoid the broad exception here and catch specific ones instead?

Reply:
Any advice on what to catch, given that we're loading code written by the user?
1. Maybe we just need to be informative?:
```
        except Exception as e:
            QgsMessageLog.logMessage(self.tr('Failed to load: {0}\n{1}').format(extpath, str(e)), 'Processing', Qgis.Critical)
            pass
```
2. If we print the exception, are you happy with `pass` rather than `raise e`?  This means if someone has written a working algorithm description, in most cases it will be useable even if they are struggling with getting an ext module module they're writing to work.
3. Should Qgis.Critical be downgraded?

Reply:
I'm fine with a QgsMessage.

Reply:
I've added that to the pull request.

Reply:
I did wonder if we should add something like this, but it didn't seem right when we don't show a message even when we fail to load an algorithm:
```
            iface.messageBar().pushMessage("Warning", "GRASS algorithm " + self.name() + " failed to load ext module - see Processing log messages", 1)